1 Niagara 4 Overview Exercise - BFile/myFile-rt/src/com/tridiumuniversity/myFile/BFile.java

```diff
@@ -0,0 +1,220 @@
+package com.tridiumuniversity.myFile;
+
+import javax.baja.nre.annotations.NiagaraAction;
+import javax.baja.nre.annotations.NiagaraProperty;
+import javax.baja.nre.annotations.NiagaraType;
+import javax.baja.sys.*;
+
+@NiagaraType
+@NiagaraProperty(
+        name = "filename",
+        type = "String",
+        flags = Flags.READONLY | Flags.SUMMARY
+)@NiagaraProperty(
+        name = "path",
+        type = "String",
+        flags = Flags.READONLY | Flags.SUMMARY
+)@NiagaraProperty(
+        name = "size",
+        type = "int",
+        flags = Flags.READONLY | Flags.SUMMARY
+)
+@NiagaraProperty(
+        name = "createdDate",
+        type = "baja:AbsTime",
+        defaultValue = "BAbsTime.DEFAULT",
+        flags = Flags.READONLY | Flags.SUMMARY
+)@NiagaraProperty(
+        name = "modifiedDate",
+        type = "baja:AbsTime",
+        defaultValue = "BAbsTime.DEFAULT",
+        flags = Flags.READONLY | Flags.SUMMARY
+)
+@NiagaraAction(
+        name="touch",
+        flags = Flags.SUMMARY
+)@NiagaraAction(
+        name="print",
+        flags = Flags.SUMMARY
+)
+
+public class BFile extends BComponent {
+//region /*+ ------------ BEGIN BAJA AUTO GENERATED CODE ------------ +*/
+//@formatter:off
+/*@ $com.tridiumuniversity.myFile.BFile(4153978362)1.0$ @*/
+/* Generated Thu Aug 01 11:29:47 KST 2024 by Slot-o-Matic (c) Tridium, Inc. 2012-2024 */
+
+  //region Property "filename"
+
+  /**
+   * Slot for the {@code filename} property.
+   * @see #getFilename
+   * @see #setFilename
+   */
+  public static final Property filename = newProperty(Flags.READONLY | Flags.SUMMARY, "", null);
```

Comment:
It looks like you may have manually edited these `Property` lines? For instance, if I run slotomatic on your code, it replaces this line with:
```java
public static final Property filename = newProperty(Flags.READONLY | Flags.SUMMARY, null, null);
```
which doesn't compile. You should be specifying a `defaultValue` in the `@NiagaraProperty` annotations above and not manually editing the code that slotomatic generates for you.